### ISSUE TITLE
spirv-builder 0.10 not on crates.io

### ISSUE THREAD

**Issue by ArrEssJay** [Rust-GPU/rust-gpu#39]
### Expected Behaviour
Latest published `spirv-builder` crate on crates.io is 0.9 - this requires toolchain `nightly-2023-05-27`, which is inconsistent with `nightly-2024-04-24` currently referred to in rust-gpu top level `rust-toolchain.toml`.

This appears to be resolved in `spirv-builder` 0.10, however this is not published.

**Comment by LegNeato** [comment]
Ah, we should push a new version every time we bump on master as folks may be using builder directly via cargo. Until it is published you can use a git dependency. Let me know if for some reason that doesn't work.

**Comment by utensil** [comment]
Sorry to barge in. I have a related question:

From the wgpu runner examples of rust-gpu, and from how GraphiteEditor handle it, one can see that the building of rust shader crates into SPIRV is done via a standalone CLI program (GraphiteEditor [made a lightweight wrapper](https://github.com/GraphiteEditor/Graphite/tree/c3a3c4c907419e2ebb9a382546a1d7b66f57cdc0/node-graph/gpu-compiler/gpu-compiler-bin-wrapper) around it to call the CLI from Rust to compile shader crates).

Is this the encouraged practice? I was imagining a more rust native way to integrate the host program (handles non-GPU logic, and GPU runtime and the execution of shaders) and the compilation of shader crates.

Particularly, if the host rust program need to use a newer rust nightly toolchain (maybe required by other crates in the host program) than spirv-builder (which needs significant more work to bump rust toolchain, thus moving slower), this way seems to be the only option? GraphiteEditor seems to choose to use the same rust toolchain for the host program as spirv-builder.

**Comment by LegNeato** [comment]
Yes, some projects use the required pinned nightly via `spirv-builder`, generate the .spv file, and then consume that via their host rust program using stable / whatever rust version they need for their application. One example is https://github.com/charles-r-earp/krnl I believe.

The project's goal is to track nightly so at least if folks are using nightly they can use one toolchain, and then create a tool (`cargo gpu`) to handle all the manual gyrations for other modes and reduce toil in general.

**Comment by schell** [comment]
@utensil  I also do this in [renderling/shaders](https://github.com/schell/renderling/tree/main/shaders). I prefer this setup because it makes recompilation explicit and indeed, you can keep using latest stable or nightly for the rest of your project. It also enables other workflows, like using `naga` to translate the generated SPIR-V into WGSL for web.

**Comment by utensil** [comment]
Thank you both for the explanations and pointers to examples ❤️ 

I had starred both repos before but didn't realize that they are both based on rust-gpu 😮
